### PR TITLE
[automation] Allow dashes in rule file names

### PR DESCRIPTION
**Pull request by kaikreuzer**
Fixes https://github.com/openhab/openhab-core/issues/1728

Signed-off-by: Kai Kreuzer <kai@openhab.org>

### CODE REVIEW

**Review by lolodomo — APPROVED** [review]
LGTM

File: bundles/org.openhab.core.model.script.runtime/src/org/openhab/core/model/script/runtime/internal/engine/DSLScriptEngine.java [hunk 1]
```diff
@@ -87,10 +87,9 @@ public Object eval(String script) throws ScriptException {
             if (script.stripLeading().startsWith(DSLScriptContextProvider.CONTEXT_IDENTIFIER)) {
                 String contextString = script.stripLeading().substring(
                         DSLScriptContextProvider.CONTEXT_IDENTIFIER.length(), script.stripLeading().indexOf('\n'));
-                String[] segments = contextString.split("-");
-                if (segments.length == 2) {
-                    modelName = segments[0];
-                    String ruleIndex = segments[1];
+                if (contextString.contains("-")) {
+                    modelName = contextString.substring(0, contextString.lastIndexOf('-'));
+                    String ruleIndex = contextString.substring(contextString.lastIndexOf('-') + 1);
```
Comment (cweitkamp) [inline bundles/org.openhab.core.model.script.runtime/src/org/openhab/core/model/script/runtime/internal/engine/DSLScriptEngine.java:92]: We can store `contextString.lastIndexOf('-')` in a local variable and reuse it:

```suggestion
                int index = contextString.lastIndexOf('-');
                if (index != -1) {
                    modelName = contextString.substring(0, index);
                    String ruleIndex = contextString.substring(index + 1);
```

**Review by cweitkamp — COMMENTED** [review]
Thanks. One minor comment for an improvement.

File: bundles/org.openhab.core.model.script.runtime/src/org/openhab/core/model/script/runtime/internal/engine/DSLScriptEngine.java (hunk 1 above)
Comment (kaikreuzer) [reply bundles/org.openhab.core.model.script.runtime/src/org/openhab/core/model/script/runtime/internal/engine/DSLScriptEngine.java]: addressed

File: bundles/org.openhab.core.model.script.runtime/src/org/openhab/core/model/script/runtime/internal/engine/DSLScriptEngine.java [hunk 2]
```diff
@@ -87,10 +87,10 @@ public Object eval(String script) throws ScriptException {
             if (script.stripLeading().startsWith(DSLScriptContextProvider.CONTEXT_IDENTIFIER)) {
                 String contextString = script.stripLeading().substring(
                         DSLScriptContextProvider.CONTEXT_IDENTIFIER.length(), script.stripLeading().indexOf('\n'));
-                String[] segments = contextString.split("-");
-                if (segments.length == 2) {
-                    modelName = segments[0];
-                    String ruleIndex = segments[1];
+                if (contextString.contains("-")) {
+                    int indexLastDash = contextString.lastIndexOf('-');
```
Comment (cweitkamp) [inline bundles/org.openhab.core.model.script.runtime/src/org/openhab/core/model/script/runtime/internal/engine/DSLScriptEngine.java:91]: Thanks. But I could be improved a little bit more. It is possible to skip the `contains("-")` check when calculation the `indexLastDash = lastIndexOf("-")` first and check for `-1` on it.

Comment (kaikreuzer) [reply bundles/org.openhab.core.model.script.runtime/src/org/openhab/core/model/script/runtime/internal/engine/DSLScriptEngine.java:91]: I preferred it this way as it is imho better readable and the few cpu cycles should not really matter :-)

Comment (cweitkamp) [reply bundles/org.openhab.core.model.script.runtime/src/org/openhab/core/model/script/runtime/internal/engine/DSLScriptEngine.java:91]: Okay. Than I will agree on it too.

**Review by cweitkamp — APPROVED** [review]
LGTM.

**Comment by cweitkamp** [comment]
The Jenkins failure looks unrelated.

**Pull request merged**